[shayaansultan] iP

shayaansultan opened the pull request:
# Bot: Your Personal Tasks, Deadlines and Events Tracker

> "The future depends on what you do today." - Mahatma Gandhi

**Bot** is a simple and efficient tool designed to free your mind from remembering all the tasks you need to do. It's:

- Text-based
- Easy to learn
- Super fast to use

## Getting Started

1. Download it from [here](https://github.com/shayaansultan/ip).
2. Run it
3. Add your tasks.
4. Let it manage your tasks for you. :wink:

And it is **FREE**!

## Features

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

## Usage

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Bot bot = new Bot();
        bot.addTask("Read book");
        bot.addDeadline("Submit report", "2021-12-01");
        bot.addEvent("Team meeting", "2021-12-02 10:00");
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

shayaansultan committed:
Level-0: Greet and Exit

shayaansultan committed:
Level-1: Echo and Exit

shayaansultan committed:
Remove class files from repository

shayaansultan committed:
Level-2: Add and List with TaskList class

shayaansultan committed:
Level-3: Mark as Done

shayaansultan committed:
Level-4: ToDos, Events, Deadlines

shayaansultan committed:
A-TextUiTesting: Implement text UI testing

shayaansultan committed:
Completed Level 5, Errors

shayaansultan committed:
Level-6 Completed, Added Delete

shayaansultan committed:
Improved error handling for Level 5

shayaansultan committed:
Created Storage Class

shayaansultan committed:
Reformat strings for Tasks for easier parsing for Storage

shayaansultan committed:
Add error handling for folder and/or file not existing

shayaansultan committed:
Add fromString for Task Classes

shayaansultan committed:
Completed Level 7

shayaansultan committed:
Merge Level 7 changes

shayaansultan committed:
Add datetime support for Event and Deadline classes

shayaansultan committed:
Merge Level 8 changes

shayaansultan committed:
Create Ui and Storage classes

shayaansultan committed:
Refactor classes to make use of Ui and Parser classes

shayaansultan committed:
Merge branch-A-MoreOOP

shayaansultan committed:
Move files into duke package

shayaansultan committed:
Merge branch-A-Packages

shayaansultan committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

shayaansultan committed:
Modify build.gradle

shayaansultan committed:
Merge branch-A-Gradle

shayaansultan committed:
Add ToDoTest.java

shayaansultan committed:
Add ParserTest.java with JUnit tests for Parser.java

shayaansultan committed:
Merge branch-A-JUnit

shayaansultan committed:
Add JavaDoc headers to Duke.java

shayaansultan committed:
Add parseDeadlineFromStorage method

shayaansultan committed:
Add JavaDoc headers to Deadline.java

shayaansultan committed:
Add Javadoc headers to DukeException.java

shayaansultan committed:
Add JavaDoc headers to Event.java

shayaansultan committed:
Add JavaDoc headers to public methods in Handler.java

shayaansultan committed:
Add new lines in print messages

shayaansultan committed:
Add JavaDoc headers to Parser.java

shayaansultan committed:
Add JavaDoc comments to Storage class

shayaansultan committed:
Add JavaDoc headers to Task class

shayaansultan committed:
Add JavaDoc headers in TaskList class

shayaansultan committed:
Add JavaDoc headers in ToDo class

shayaansultan committed:
Refactor Ui class and add JavaDoc headers

shayaansultan committed:
Refactor code to comply with coding standard

shayaansultan committed:
Add Find feature to find tasks by searching for a keyword

shayaansultan committed:
Merge branch-A-JavaDoc

shayaansultan committed:
Merge branch-A-CodingStandard

shayaansultan committed:
Merge branch-Level-9

AidenLYT commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,69 @@
+    @Override
+    public String toString() {
+        return "D | " + super.toString() + " | "
+                + by.format(DateTimeFormatter.ofPattern("MMM dd yyyy hh:mma"));
```
Good awareness of Layout practices 👍🏻

AidenLYT commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,74 @@
+            this.from = LocalDateTime.parse(from,
+                    DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm"));
+            this.to = LocalDateTime.parse(to,
+                    DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm"));
```
Great awareness of good Layout practice 👍🏻

AidenLYT commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,74 @@
+     * @param input The string to create the Event object from.
+     * @return A new Event object, or null if the Event object could not be created.
+     */
+    public static Event fromString(String input) {
```
fromString might not be a good naming for creating a new Event. Naming is quite ambiguous 😢

AidenLYT commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,84 @@
+     * @param input The user input.
+     * @return An array of two strings.
+     */
+    public String[] parse(String input) {
```
Naming could be better 😅 
Might be ambiguous to differ between parse and other parse variations

AidenLYT commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,73 @@
+     * @param input The string to parse.
+     * @return A new Task object.
+     */
+    public static Task fromString(String input) {
```
Naming could be more descriptive that the method is creating a Task

AidenLYT commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,164 @@
+    public void deleteTask(int index) throws DukeException {
+        ui.printMessage("Noted. I've removed this task: \n" + tasks.get(index - 1).toString()
+                + "\nNow you have " + (tasks.size() - 1) + " tasks in the list.\n");
+        tasks.remove(index - 1);
```
Suggestion: could initialize another int object for index - 1, so that you don't have to use 'index - 1' in full form twice. It could also improve readability if you were to give a name for index - 1 (why did you do this, what does this mean?)

AidenLYT commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,164 @@
+        List<Task> foundTasks = new ArrayList<>();
+
+        for (Task task : tasks) {
+            if (task.getDescription().toLowerCase().contains(keyword.toLowerCase())) {
```
for readability, can consider initializing 
Obj1 = task.getDescription().toLowerCase() 
and 
Obj2 = keyword.toLowerCase
before passing them to the if-statement

AidenLYT reviewed (commented):
LGTM 👍🏻 Code is easy to read and follow. Just a few nits to fix. There are a few naming suggestions to consider. 
Overall, great use of OOP and Abstraction. Well-aware of layout coding practice as well, impressed!

shayaansultan committed:
Update build.gradle to add JavaFX dependencies

shayaansultan committed:
Add JavaFX GUI following SE-EDU tutorial till part 3 iteration 1

shayaansultan committed:
Make scrollPane reactive

shayaansultan committed:
Add images for chatbot

shayaansultan committed:
Add Iteration 2

shayaansultan committed:
Add Iteration 3 of JavaFX GUI tutorial

shayaansultan committed:
Move code from Gui to Duke for refactoring and create MainWindow Class

shayaansultan committed:
Refactor code for readability

shayaansultan committed:
Add GUI Functionality

shayaansultan committed:
Remove unnecessary comments

shayaansultan committed:
Change bot image

shayaansultan committed:
Merge Level 10

shayaansultan committed:
Update application mainClass to duke.Launcher

The mainClass in build.gradle was updated from duke.Duke to duke.Launcher.
This change was necessary because the application is now a JavaFX app,
which needs to be launched from the duke.Launcher class.

Updating the mainClass ensures that the application starts correctly when
built using Gradle. This change does not affect the functionality of the
application, but it is crucial for the build process.

shayaansultan committed:
Add assertions in Handler.java

shayaansultan committed:
Add assertions in Deadline.java

shayaansultan committed:
Add assertions in Parser.java

shayaansultan committed:
Add assertions in Storage.java

shayaansultan committed:
Fix unneccessary text clipping

shayaansultan committed:
Rewrite header to follow coding standard

shayaansultan committed:
Merge branch 'master' into branch-A-CodeQuality

shayaansultan committed:
Revert "Fix unneccessary text clipping"

This reverts commit ab857a7073ec9aea426864ea90b1374ff5f9f7f3.

shayaansultan committed:
Revert "Revert "Fix unneccessary text clipping""

This reverts commit 8d66d89645dda46dab6dd471bd878367fcf16883.

shayaansultan committed:
Add assertions to TaskList class

There were several methods in the TaskList class that did not have
assertions to check the assumptions about the state of the program.

Adding assertions improves the robustness of the code by catching
unexpected behavior early.

In this commit, assertions were added to the addTask, deleteTask,
markTaskAsDone, and unmarkTaskAsDone methods in the TaskList class.
These assertions check that tasks are correctly added and removed
from the task list, and that task indices are within the valid range.

shayaansultan committed:
Merge pull request #2 from shayaansultan/branch-A-Assertions

shayaansultan committed:
Merge pull request #3 from shayaansultan/branch-A-CodeQuality

shayaansultan committed:
Add help functionality

The bot previously lacked a help functionality, which made it difficult
for users to understand how to use the various features of the bot.

In this commit, a getHelpString method was added to the TaskList class.
This method returns a string that describes all the features of the
TaskList class and how to use them. The features covered include adding
tasks, adding deadlines, displaying tasks, deleting tasks, marking tasks
as done, unmarking tasks as done, and finding tasks. This method is then
called by the input handler.

Adding this help functionality improves the user-friendliness of the bot
by providing users with a quick way to learn how to use the bot's features.

shayaansultan committed:
Merge branch-C-Help

shayaansultan committed:
Rename all Duke references to Bot

shayaansultan committed:
Add Ui.png

shayaansultan committed:
Add User Guide

shayaansultan committed:
Update build config

shayaansultan committed:
Update version to Java 11

shayaansultan committed:
Update getHelpString to align with functionality

shayaansultan committed:
Update tests in ParserTest to account for DukeExceptions

shayaansultan committed:
Add additional error handling

shayaansultan committed:
Update parser and handler to properly pass events and add basic error handling

shayaansultan committed:
Update README.md to align with Event functionality
